frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,167 @@
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton } from '../common';
+
+export default function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [namespaceNameValid, setNamespaceNameValid] = useState(false);
+  const [nameHelperMessage, setNameHelperMessage] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+  const dispatch = useDispatch();
+
+  function createNewNamespace() {
+    const clusterData = getCluster();
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+        creationTimestamp: new Date().toISOString(),
+        uid: '',
+      },
+    };
+
+    const newNamespaceName = newNamespaceData.metadata.name;
+
+    async function namespaceRequest() {
+      try {
+        const response = await Namespace.apiEndpoint.post(newNamespaceData, {}, clusterData || '');
+        return response;
+      } catch (error: any) {
+        const statusCode = error?.status;
+        console.error('Error creating namespace:', error);
+        if (statusCode === 409) {
+          setNamespaceDialogOpen(true);
+          setNamespaceNameValid(false);
+          setNameHelperMessage(t('translation|A namespace with this name already exists.'));
+        }
+        throw error;
+      }
+    }
+
+    setNamespaceDialogOpen(false);
+
+    dispatch(
+      clusterAction(() => namespaceRequest(), {
+        startMessage: t('translation|Applying {{  newItemName  }}…', {
+          newItemName: newNamespaceName,
+        }),
+        cancelledMessage: t('translation|Cancelled applying  {{  newItemName  }}.', {
+          newItemName: newNamespaceName,
+        }),
+        successMessage: t('translation|Applied {{ newItemName }}.', {
+          newItemName: newNamespaceName,
+        }),
+        errorMessage: t('translation|Failed to create {{ kind }} {{ name }}.', {
+          kind: 'namespace',
+          name: newNamespaceName,
+        }),
+        cancelCallback: () => {
+          setNamespaceDialogOpen(true);
+        },
+      })
+    );
+  }
+
+  useEffect(() => {
+    setNamespaceNameValid(Namespace.isValidNamespaceFormat(namespaceName));
+
+    if (!Namespace.isValidNamespaceFormat(namespaceName)) {
+      if (namespaceName.length > 63) {
+        setNameHelperMessage(t('translation|Namespaces must be under 64 characters.'));
+      } else {
+        setNameHelperMessage(
+          t(
+            "translation|Namespaces must contain only lowercase alphanumeric characters or '-', and must start and end with an alphanumeric character."
+          )
+        );
+      }
+    }
+  }, [namespaceName]);
+
+  return (
+    <>
```

Comment:
The name needs to be wrapped in a AuthVisible component, because we don't want to show it unless the user can create the namespace.